pkg/fieldpath/path.go

```diff
@@ -92,6 +101,18 @@ func (p *Path) Copy() *Path {
 	return &Path{p.parts}
 }
 
+// CopyAt returns a new Path that is a copy of this Path up to the supplied
+// index.
+//
+// e.g. given Path $A containing "X.Y", $A.CopyAt(0) would return a new Path
+// containing just "X". $A.CopyAt(1) would return a new Path containing "X.Y".
+func (p *Path) CopyAt(index int) *Path {
+	if index < 0 || len(p.parts) < 0 || index > len(p.parts)-1 {
```

Comment:
nit: I think length function will never return less than 0. You can just check for empty i.e." == 0"
```suggestion
	if index < 0 || len(p.parts) == 0 || index > len(p.parts)-1 {
```